File: backend/app/schemas/achievement.py
from datetime import datetime
from pydantic import BaseModel


class AchievementResponse(BaseModel):
    id: int
    name: str
    description: str
    badge_emoji: str
    badge_color: str
    condition_type: str
    condition_module_id: int | None = None
    is_earned: bool = False
    earned_at: datetime | None = None

    model_config = {"from_attributes": True}

File: frontend/src/types/achievement.ts
export type AchievementConditionType =
  | 'first_topic'
  | 'module_completed'
  | 'streak_days'
  | 'chat_messages'
  | 'quiz_perfect'
  | 'course_completed'

export interface Achievement {
  id: number
  name: string
  description: string
  badge_emoji: string
  badge_color: string
  condition_type: AchievementConditionType | string
  condition_module_id: number | null
  is_earned: boolean
  earned_at: string | null
}
